src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Telemetry/VsTelemetryService.cs

```diff
@@ -0,0 +1,238 @@
+                {
+                    var configuredProject = await unconfiguredProject.GetSuggestedConfiguredProjectAsync().ConfigureAwait(false);
+                    var properties = configuredProject.Services.ProjectPropertiesProvider.GetCommonProperties();
+
+                    List<string> propertyNames = new List<string>()
+                    {
+                        ConfigurationGeneral.TargetFrameworkProperty,
+                        ConfigurationGeneral.TargetFrameworksProperty,
+                        ConfigurationGeneral.OutputTypeProperty
+                    };
+
+                    foreach (var propertyName in propertyNames)
+                    {
+                        var propertyValue = await properties.GetEvaluatedPropertyValueAsync(propertyName).ConfigureAwait(false);
+                        if (!string.IsNullOrEmpty(propertyValue))
+                        {
+                            projectProperties[BuildPropertyName(eventName, propertyName)] = propertyValue;
+                        }
+                    }
+                }
+
+                var projectAsset = TelemetryService.DefaultSession.PostAsset(
+                    EventPrefix + eventName.ToLowerInvariant(),
+                    assetId,
+                    0,
+                    projectProperties);
+
+                projectAssets.Add(assetId, projectAsset);
+            }
+        }
+
+        /// <summary>
+        /// Build a fully qualified property name based on it's parent event and the property name
+        /// </summary>
+        /// <param name="eventName">Name of the parent event.</param>
+        /// <param name="propertyName">Name of the property.</param>
+        /// <returns>Fully qualified property name.</returns>
+        /// <remarks>
+        /// Properties are expected to be in the following format - EventName.PropertyName
+        /// with the the slashes from the vent name replaced by periods.
+        /// e.g. vs/myevent would translate to VS.MyEvent.MyProperty
+        /// </remarks>
+        private string BuildPropertyName(string eventName, string propertyName)
+        {
+            string name = "VS.ProjectSystem.Managed." + eventName.Replace('/', '.');
+
+            if (!name.EndsWith("."))
+            {
+                name += ".";
+            }
+
+            name += propertyName;
+
+            return name;
+        }
+
+        /// <summary>
+        /// Tries to correlate a project 
+        /// </summary>
+        /// <param name="unconfiguredProject">Correlated project asset for this event.</param>
+        /// <param name="telemetryEvent">Telemetry event to correlate the asset to.</param>
+        private void TryCorrelateProject(UnconfiguredProject unconfiguredProject, TelemetryEvent telemetryEvent)
+        {
+            if (unconfiguredProject != null && projectAssets.TryGetValue(GetProjectId(unconfiguredProject), out TelemetryEventCorrelation projectAsset))
```

Comment:
hmmm.. but dictionary is mutated. which means, if it is used concurrently, it can get into corrupted state.